agent.go

```diff
@@ -121,6 +121,7 @@ type Agent struct {
 	log           logging.LeveledLogger
 
 	net *vnet.Net
+	tcp *tcpIPMux
```

Comment:
What do you think of making `tcpIPMux` always non-nil?

`if a.tcp != nil` kind of scares me. I can see new code getting added that misses that check and would crash in ways we would fail to test.

Reply:
Makes sense, changed!